exist-core/src/main/java/org/exist/dom/QName.java

```diff
@@ -205,15 +202,14 @@ public int compareTo(final QName other) {
      */
     @Override
     public boolean equals(final Object other) {
-        if(other == this) {
+        if (other == this) {
             return true;
-        } else if(!(other instanceof QName)) {
+        }
+        if (!(other instanceof final QName qnOther)) {
             return false;
-        } else {
-            final QName qnOther = (QName)other;
-            return this.namespaceURI.equals(qnOther.namespaceURI)
-                    && this.localPart.equals(qnOther.localPart);
         }
+        return this.namespaceURI.equals(qnOther.namespaceURI)
```

Comment:
I find this a little tricky to read. Is `qnOther` in-scope here? I know they made some changes to `instance of` in newer Java versions so that you can bind variables, but just reading this code, I would expect `qnOther` to be out-of-scope as it is within the context of the `if` statement and this `return` argument is in a separate scope.

Reply:
I find this tricky as well, I just gave in as the IDE kept insisting this is the way to go. It can be easily refactored to be way easier to read. Maybe there is a setting to ignore this "optimisation" in IntelliJ.

Reply:
qnOther must be in scope as all tests pass and I strongly suspect that QName equality is covered by several test cases.

Reply:
As it is difficult for both of us to read. I suggest we reverse this small change. Automated suggestions are not always correct/best

Reply:
Please have a look at the current form. This is now quite obvious.